src/Functions/hasAnyAllTokens.cpp

```diff
@@ -66,8 +67,18 @@ void FunctionHasAnyAllTokens<SearchTraits>::setSearchTokens(const std::vector<St
         throw Exception(ErrorCodes::BAD_ARGUMENTS, "Function '{}' supports a max of {} needles", name, supported_number_of_needles);
 }
 
-template <class SearchTraits>
-DataTypePtr FunctionHasAnyAllTokens<SearchTraits>::getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const
+namespace
+{
+
+bool isStringArray(const IDataType & type)
+{
+    return isArray(type) && isString(checkAndGetDataType<DataTypeArray>(type).getNestedType());
```

Comment:
I suppose it does, but this test will work also for nullable strings?

Reply:
`toNullable(['token'])` is forbidden anyways (values of type `Nullable` must not contain nested types).

`[toNullable('token')]` gives the correct error message.